src/Datadog.AutoInstrumentation.NativeLoader/cor_profiler.cpp

```diff
@@ -0,0 +1,4587 @@
+        return std::atomic_fetch_add(&this->m_refCount, 1) + 1;
+    }
+
+    ULONG STDMETHODCALLTYPE CorProfiler::Release(void)
+    {
+        Debug("CorProfiler::Release");
+        int count = std::atomic_fetch_sub(&this->m_refCount, 1) - 1;
+
+        if (count <= 0)
+        {
+            delete this;
+        }
+
+        return count;
+    }
+
+
+    HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* pICorProfilerInfoUnk)
+    {
+        //
+        // Get and set profiler pointers
+        //
+        if (m_dispatcher == nullptr)
+        {
+            return E_FAIL;
+        }
+        DynamicInstance* cpInstance = m_dispatcher->GetContinuousProfilerInstance();
+        if (cpInstance != nullptr)
+        {
+            m_cpProfiler = cpInstance->GetProfilerCallback();
+        }
+        DynamicInstance* tracerInstance = m_dispatcher->GetTracerInstance();
+        if (tracerInstance != nullptr)
+        {
+            m_tracerProfiler = tracerInstance->GetProfilerCallback();
+        }
+        DynamicInstance* customInstance = m_dispatcher->GetCustomInstance();
+        if (customInstance != nullptr)
+        {
+            m_customProfiler = customInstance->GetProfilerCallback();
+        }
+
+        //
+        // Get Profiler interface ICorProfilerInfo6 for net46+
+        //
+        ICorProfilerInfo6* info6 = nullptr;
+        HRESULT hr = pICorProfilerInfoUnk->QueryInterface(__uuidof(ICorProfilerInfo6), (void**) &info6);
+        if (FAILED(hr))
+        {
+            Warn("Failed to attach profiler: interface ICorProfilerInfo6 not found.");
+            return E_FAIL;
+        }
+
+        // Gets the initial value for the event mask
+        DWORD mask_low;
+        DWORD mask_hi;
+        hr = info6->GetEventMask2(&mask_low, &mask_hi);
+        if (FAILED(hr))
+        {
+            Warn("Error getting the event mask.");
+            return E_FAIL;
+        }
+
+        Debug("MaskLow: ", mask_low);
+        Debug("MaskHi : ", mask_hi);
+
+        //
+        // Continuous Profiler Initialization
+        //
+        if (m_cpProfiler != nullptr)
+        {
+            HRESULT localResult = m_cpProfiler->Initialize(pICorProfilerInfoUnk);
+            if (SUCCEEDED(localResult))
+            {
+                // let's get the event mask set by the CP.
+                DWORD local_mask_low;
+                DWORD local_mask_hi;
+                HRESULT hr = info6->GetEventMask2(&local_mask_low, &local_mask_hi);
+                if (SUCCEEDED(hr))
+                {
+                    mask_low = mask_low | local_mask_low;
+                    mask_hi = mask_hi | local_mask_hi;
+
+                    Debug("*LocalMaskLow: ", local_mask_low);
+                    Debug("*LocalMaskHi : ", local_mask_hi);
+                }
+                else
+                {
+                    Warn("Error getting the event mask.");
+                }
+            }
+            else
+            {
```

Comment:
In a different PR I [explained](https://github.com/DataDog/dd-continuous-profiler-dotnet/pull/117#discussion_r683693043) why I think that handling the shorter (in this case the negative) case is better. Either way, it's not something that should block anything, just FYI. I'll put a follow up tag, so that this gets noticed, but feel free to leave it as is if you prefer.

Reply:
❗ [follow up in future PR]